code/controllers/CMSMain.php

```diff
@@ -972,13 +972,9 @@ public function save($data, $form) {
 
 		// Update the class instance if necessary
 		if(isset($data['ClassName']) && $data['ClassName'] != $record->ClassName) {
-			$newClassName = $record->ClassName;
-			// The records originally saved attribute was overwritten by $form->saveInto($record) before.
-			// This is necessary for newClassInstance() to work as expected, and trigger change detection
-			// on the ClassName attribute
-			$record->setClassName($data['ClassName']);
-			// Replace $record with a new instance
-			$record = $record->newClassInstance($newClassName);
+            // Replace $record with a new instance of the new class
+			$newClassName = $data['ClassName'];
+            $record = $record->newClassInstance($newClassName);
```

Comment:
consistent tabbing please

Reply:
Odd, indentation looked fine on my fork (and IDE), guessing there's a repo setting for that I don't know about. Anyway, a few odd commits later, and it should be fine now.